src/hclib-promise.c

```diff
@@ -179,6 +179,16 @@ int register_on_all_promise_dependencies(hclib_task_t *wrapper_task) {
         }
     }
 
+    if (wrapper_task->waiting_on_extra) {
+        while (wrapper_task->waiting_on_extra[wrapper_task->waiting_on_index - MAX_NUM_WAITS + 1]) {
+            wrapper_task->waiting_on_index++;
+            hclib_future_t *curr = wrapper_task->waiting_on_extra[wrapper_task->waiting_on_index - MAX_NUM_WAITS];
+            if (_register_if_promise_not_ready(wrapper_task, curr)) {
```

Comment:
should we add the  if (curr) check as above since nullptr is allowed to be passed as dependency?

Reply:
I'm not sure I would really consider that desirable behavior, and would actually break the code in this case as it is assuming that the list is NULL-terminated. This is nice as it means we don't have to add another field to the task structure and have it take up more space for tasks that don't use futures.

We could still add a field that is the # of futures. But a better question is, do you know if any of our benchmarks or tests currently assume they can pass a NULL pointer as part of the future list and it'll just be ignored? Do you disagree, and think that we should support this?

Reply:
In that case, let us keep it as a null-terminated list so that we do not have to take the additional overhead of # of futures field. If we specify it is a null-terminated list, then there is no confusion, and it is clear there cannot be nullptr in between. I do not know of any benchmark that would need such a feature of having nullptr in the middle.